Realm/RLMGeospatial.mm

```diff
@@ -0,0 +1,232 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2023 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#import "RLMGeospatial_Private.hpp"
+#import "RLMUtil.hpp"
+
+#import <realm/geospatial.hpp>
+@implementation RLMGeospatialPoint
+- (nullable instancetype)initWithLatitude:(double)latitude longitude:(double)longitude {
+    return [self initWithLatitude:latitude longitude:longitude altitude:0];
+}
+
+- (nullable instancetype)initWithLatitude:(double)latitude longitude:(double)longitude altitude:(double)altitude {
+    if (self = [super init]) {
+        if ((latitude < -90 || latitude > 90) || (longitude < -180 || longitude > 180) || (altitude < 0)) {
```

Comment:
Since we're explicitly rejecting out-of-bounds values we probably also want to reject NaN, and as written this'll currently accept NaN for each of the fields (comparisons against NaN always return false).